pdr_backend/pdr_dashboard/dash_components/plots.py

```diff
@@ -524,43 +549,59 @@ def get_figures_and_metrics(
 
 @enforce_types
 def get_feed_figures(
-    payouts: pandas.DataFrame, subscriptions: pandas.DataFrame
+    payouts: pl.DataFrame, subscriptions: pl.DataFrame
 ) -> FeedModalFigures:
     """
     Return figures for a selected feed from the feeds table.
     """
     # Initialize empty figures with default settings
     result = FeedModalFigures()
 
-    if payouts.empty:
+    if payouts.is_empty():
         return result
 
     # Process subscription data
     result.subscription_purchases = subscriptions["count"]
     result.subscription_revenues = subscriptions["revenue"]
-    result.subscription_dates = subscriptions["day"].apply(
-        lambda x: datetime.datetime.combine(x, datetime.time())
-    )
 
-    payouts["profit"] = payouts.apply(
-        lambda x: max(x["payout"], 0) - x["stake"], axis=1
+    subscriptions = subscriptions.with_columns(
+        pl.col("day")
+        .map_elements(
+            lambda x: datetime.datetime.combine(x, datetime.time()), pl.Datetime
+        )
+        .alias("day_dt"),
+    )
+    result.subscription_dates = subscriptions["day_dt"]
+
+    payouts = payouts.with_columns(
+        pl.struct(["payout", "stake"])
+        .map_elements(lambda x: max(x["payout"], 0) - x["stake"], pl.Float64)
```

Comment:
why do we map here, what is the diff with the following code?


```
payouts = payouts.with_columns(
    [
        (pl.col("payout").clip_min(0) - pl.col("stake")).alias("profit"),
        pl.lit(1).alias("count"),
        pl.lit(1).cum_sum().alias("cnt_cumsum"),
        pl.when(pl.col("payout") > 0)
        .then(1)
        .otherwise(0)
        .cum_sum()
        .alias("cnt_corrpred")
    ]
)
```